common/app/rendering/core/JavascriptProps.scala

```diff
@@ -3,21 +3,60 @@ package rendering.core
 import conf.Configuration
 import play.api.libs.json.{JsValue, Json, Writes}
 
-case class JavascriptPropsConfig(bundleUrl: String, polyfillioUrl: String, beaconUrl: String)
+case class TypeFace(typeFace: String, fileTypes: Seq[FileType])
+
+object TypeFace {
+  implicit val typeFaceWriter: Writes[TypeFace] = Json.writes[TypeFace]
+}
+
+case class FileType(fileType: String, endpoint: String, hintTypes: Seq[HintType])
+
+object FileType {
+  implicit val fileTypeWriter: Writes[FileType] = Json.writes[FileType]
+}
+
+case class HintType(hintType: String, endpoint: String)
+
+object HintType {
+  implicit val hintTypeWriter: Writes[HintType] = Json.writes[HintType]
+}
+
+case class JavascriptPropsConfig(bundleUrl: String, polyfillioUrl: String, beaconUrl: String, fontDefinitions: JsValue)
 case class JavascriptProps(config: JavascriptPropsConfig){
   def asJsValue: JsValue = Json.toJson(this)
 }
 
 object JavascriptProps {
   implicit val javascriptPropsConfig: Writes[JavascriptPropsConfig] = Json.writes[JavascriptPropsConfig]
   implicit val javascriptProps: Writes[JavascriptProps] = Json.writes[JavascriptProps]
+
+  def getFontDefinitions(): JsValue = {
+    val typeFaces = List("GuardianEgyptianWeb", "GuardianTextEgyptianWeb", "GuardianSansWeb", "GuardianTextSansWeb")
+    val fileTypes = List("woff2", "woff", "ttf")
+    val hintTypes = List("cleartype", "auto")
+
+    val fontDefinitions = typeFaces.map(typeFace => {
+      TypeFace(typeFace, fileTypes.map { fileType =>
+        val fileTypeEndpoint = conf.Static(s"fonts/${typeFace}.${fileType}.json")
+        
+        FileType(fileType, fileTypeEndpoint, hintTypes.map { hintType => {
+          val hintTypeEndpoint = conf.Static(s"fonts/${typeFace}${hintType.capitalize}Hinted.${fileType}.json")
+          HintType(hintType, hintTypeEndpoint)
+        }})
+      })
+    })
+
+    Json.toJson(fontDefinitions)
```

Comment:
Since there's an `asJsValue` method on `JavascriptProps`, I don't think it is necessary to do this conversion to JSON here. Instead, the `fontDefinitions` parameter for `JavascriptPropsConfig` could be a `List[TypeFace]`... it will be implicitly converted to JSON (if necessary) when a dev calls `JavascriptProps.asJsValue()`

What do you think?

Reply:
Makes sense, I've made the changes as suggested